### PR TITLE
storage: change low-level liveness methods to regular time 

### PR DESCRIPTION
**Pull request by andreimatei**
liveness.IsLive/IsDead were taking an HLC, although liveness is
fundamentally a physical-clock-related concept.

Release note: None

### CODE REVIEW

File: pkg/jobs/registry.go [hunk 1]
```diff
@@ -741,7 +741,7 @@ func (r *Registry) maybeAdoptJob(ctx context.Context, nl NodeLiveness) error {
 			// Don't try to start any more jobs unless we're really live,
 			// otherwise we'd just immediately cancel them.
 			if liveness.NodeID == r.nodeID.Get() {
-				if !liveness.IsLive(r.clock.Now()) {
+				if !liveness.IsLive(r.clock.Now().GoTime()) {
```
Comment (tbg) [inline pkg/jobs/registry.go]: PhysicalNow

**Review by tbg — APPROVED** [review]
I tried to come up with a better rationale for the commit message but it was difficult. Maybe

> Liveness previously used hlc timestamps, but there is no compelling reason to do so. We use liveness to reason about the availability of nodes typically via an argument of the form "since the liveness expiration plus MaxOffset is in the past of my local clock, this liveness is also expired on the owner". Remove usage of hlc to simplify things.

File: pkg/jobs/registry.go (hunk 1 above)
Comment (andreimatei) [reply pkg/jobs/registry.go]: actually, I switched everybody the other way. See note at the top.

**Comment by andreimatei** [comment]
> I tried to come up with a better rationale for the commit message but it was difficult

Mmmm, I didn't like your wording very much :). I've amended mine to cite the fact that liveness record expiration did not have a logical component as reason enough for this change.

I've also done another change throughout the patch, so please take another look - I've switched all the users of these API from `clock.PhysicalTime()` to `clock.Now().GoTime()`. The latter takes into account (higher) clock signals coming from other nodes, which means that, if the clock synchronization built into CRDB works well, different nodes will converge quicker on agreeing that a particular liveness record is expired. This matches most of the old behavior, although the code used to be inconsistent on this point.
This seems like the right thing to do to me, but please tell me if you have thoughts.

**Review by tbg — COMMENTED** [review]
Looking at this again, I don't know. The important thing in this change is justifying why it should be one or the other. Now we're mixing both and not giving a good rationale on which it should be. For example, why aren't we also taking `c.Now().GoTime()` going forward as we did prior to this PR? I'd think we either throw out the hlc to keep things simple or we use it everywhere.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43861#-:-Lyd_RshCXSW3CUlzTas:b-ha8x79)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @nvanbenschoten)

**Review by tbg — COMMENTED** [review]
> Reviewed 3 of 8 files at r1, 1 of 2 files at r2, 7 of 7 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43861)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @nvanbenschoten)

**Review by andreimatei — COMMENTED** [review]
> For example, why aren't we also taking c.Now().GoTime() going forward as we did prior to this PR?

What do you mean? "Taking" it where?

We are throwing out HLC. We're not mixing both - at least I don't see it that way.
But having thrown out HLC, the question about "what time is it" still remains. The way I see it, there's multiple clocks one can look at - as always. One is `c.PhysicalTime()`, the other is `c.Now().GoTime()`. The former has the property that time flows at a constant rate (more or less). The latter has the property that it's closer to the other values in the cluster. I figured that in the case of the liveness record, which acts like a sort of lease, the consistency property of `c.Now().GoTime()` is more important. Basically, `c.Now().GoTime()` is a better clock because it takes into consideration the clock syncing done both by NTP and by CRDB.
Another argument for using `c.Now().GoTime()` is that `c.Now()` is used when setting `l.Expiration`. If we went the other way, I guess we should change that too.
https://github.com/cockroachdb/cockroach/blob/bc5a9f13e4e3d8d1fee16b602f7374f379a4c2ed/pkg/storage/node_liveness.go#L577

You buy?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43861#-:-Lydi2YT5YnU8XvjFm4g:b-v5rvvw)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @nvanbenschoten)

___
*[pkg/jobs/registry.go, line 744 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/43861#-LyBu1rv4h1KapATyZdY-r2-744:-LyZbvWvCspkU5jLPET4:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/09af7b69ea50148106550e181d263c9ad80d43bb/pkg/jobs/registry.go#L744)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

PhysicalNow
</blockquote></details>

done

**Review by tbg — COMMENTED** [review]
> What do you mean? "Taking" it where?

Oh, I actually meant the snippet about update.Expiration that you posted and had erroneously assumed that this PR switched it to using the system time. Nevermind then, this works for me. Sorry about the back and forth.

:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43861#-:-LydpSrC5GJB62dYOdrO:beb4dvi)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei and @nvanbenschoten)

**Review by tbg — COMMENTED** [review]
> Reviewed 1 of 8 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43861)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei and @nvanbenschoten)

**Review by andreimatei — COMMENTED** [review]
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43861#-:-Lz8pevLDT_Kyh_XucIm:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @andreimatei, @nvanbenschoten, and @tbg)

**Review by andreimatei — COMMENTED** [review]
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43861#-:-LzI8qbH3rtYgzatIGa_:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @andreimatei, @nvanbenschoten, and @tbg)

**Pull request merged**